File: ABC/ABC128/C.cpp
// refactoring for blog
#include <bits/stdc++.h>
#include <atcoder/all>
using namespace std;
using namespace atcoder;

#define fs(n) fixed << setprecision(n);
#define mp(a, b) make_pair(a, b);
using ll = long long;
using ld = long double;

int main(){
  int n, m; cin >> n >> m;
  vector<vector<int>> sl(m, vector<int>(0));
  for(int i = 0; i < m; i++){
    int k; cin >> k;
    for(int j = 0; j < k; j++){
      int s; cin >> s;
      sl[i].emplace_back(s - 1);
    }
  }
  vector<int> p(m);
  for(int i = 0; i < m; i++) cin >> p[i];

  int ans = 0;
  for(int i = 0; i < (1 << n); i++){
    int light = 0;
    for(int j = 0; j < m; j++){
      int res = 0;
      for(int k : sl[j]) if(i & (1 << k)) res++;
      if(res % 2 == p[j]) light++;
    }
    if(light == m) ans++;
  }
  cout << ans << endl;
}

File: ABC/ABC128/C.py
n, m = map(int, input().split())
sl = []
for i in range(m):
  sl.append([])
  inputlist = input().split()
  k = int(inputlist[0])
  for j in range(k):
    s = int(inputlist[j + 1])
    sl[i].append(s - 1)

p = list(map(int, input().split()))

ans = 0
for i in range(1 << n):
  swlist = []
  for j in range(n):
    swlist.append(1 if i & (1 << j) else 0)
  light = 0
  for j in range(m):
    res = 0
    for k in sl[j]: res += swlist[k]
    if res % 2 == p[j]: light += 1
  if light == m: ans += 1

print(ans)